server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1243,6 +1243,17 @@ protected void addScriptValidation(
             });
         }
 
+        protected void disallowCopyFields() {
+            if (copyTo.hasValues()) {
+                deprecationLogger.warn(
```

Comment:
does using the deprecationLogger here imply we want to deprecate and remove allowing (and ignoring) copy_to on these field types in the future? Or do we keep the warning behaviour for an unforseeable amount of time? I don't know if in the later case there is a way to only warn when this happens.

Reply:
I think ideally we'd change this to something stricter in the future, but while we're still working through our deprecation policy this is the way to go.